Makefile

```diff
@@ -335,7 +335,11 @@ format-rust: ## Reformat all Rust code
 
 .PHONY: format-python
 format-python: ## Reformat all Python code
-	./pants fmt ::
+	./pants filter --target-type=python_library,python_tests :: | xargs ./pants fmt
+
+.PHONY: format-shell
+format-shell: ## Reformat all shell_libraries
+	./pants filter --target-type=shell_library,shunit2_tests :: | xargs ./pants fmt
 
 .PHONY: format-prettier
 format-prettier: build-formatter ## Reformat js/ts/yaml
```

Comment:
Don't forget to add "toml" to the comment now (markdown, too, I suppose!)

Reply:
It also occurs to me that there's actually no benefit of having a `build-formatter` target, since it's not a dependency of anything else.

We could just pull that build command into this target directly and simplify the Makefile just a tiny bit. `build-formatter` is a `PHONY` target, so it's always going to run anyway.

Actually, even better than that would just be to enable Buildkit builds in Docker Compose and get rid of the explicit build step altogether, and just have this be a `docker-compose run` invocation.

Per https://www.docker.com/blog/faster-builds-in-compose-thanks-to-buildkit-support/, we'll need to set the following environment variables: `COMPOSE_DOCKER_CLI_BUILD=1 DOCKER_BUILDKIT=1`. (I tested this locally; works great.)

(Thoughts on this, @inickles-grapl?)

Reply:
(currently, lint-prettier also depends on that build step. however, I whole-heartedly agree with your docker-compose suggestion, but I think it's best suited for a separate PR that applies it universally?)

Reply:
`lint-prettier` would end up looking the same as I suggest here; `build-formatter` would just disappear completely.

Doing this in a separate PR is a good way to go, though.

Reply:
> It also occurs to me that there's actually no benefit of having a build-formatter target, since it's not a dependency of anything else.
> 
> We could just pull that build command into this target directly and simplify the Makefile just a tiny bit. build-formatter is a PHONY target, so it's always going to run anyway.

Agreed, and I had a similar thought recently about some of the the `build-test-` targets we have as well.

> Actually, even better than that would just be to enable Buildkit builds in Docker Compose and get rid of the explicit build step altogether, and just have this be a docker-compose run invocation.
> 
> Per https://www.docker.com/blog/faster-builds-in-compose-thanks-to-buildkit-support/, we'll need to set the following environment variables: COMPOSE_DOCKER_CLI_BUILD=1 DOCKER_BUILDKIT=1. (I tested this locally; works great.)

Yeah, I think that's fine as long as the builds depend on a single Dockerfile of ours, and we're not using the 'builder pattern' as described from: https://docs.docker.com/develop/develop-images/multistage-build/.